[JIT] add support for torch.lu to torchscript

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #33818 [JIT] Add support for list()
* #33783 [JIT] add support for torch.norm
* #33737 [JIT] add support for torch.cdist
* #33736 [JIT] add support for lu_unpack
* **#33724 [JIT] add support for torch.lu to torchscript**
* #33504 [JIT] fix resolving of functions in torch/functional. fix compilation of torch.stft

Fix for https://github.com/pytorch/pytorch/issues/33381, partial fix of https://github.com/pytorch/pytorch/issues/30786

Differential Revision: [D20077321](https://our.internmc.facebook.com/intern/diff/D20077321)

Commit:
add support for torch.lu to torchscript

[ghstack-poisoned]

vishwakftw commented on test/test_jit.py:
```diff
@@ -10194,6 +10194,18 @@ def foo(input, n_fft):
+            # type: (Tensor) -> Tuple[Tensor, Tensor, Tensor]
+            return torch.lu(x, get_infos=True)
+
+        self.checkScript(lu, (torch.randn(2, 3, 3),))
```
nit: this should've been `lu_infos`?

Commit:
Update on "[JIT] add support for torch.lu to torchscript"


Fix for https://github.com/pytorch/pytorch/issues/33381, partial fix of https://github.com/pytorch/pytorch/issues/30786

Differential Revision: [D20077321](https://our.internmc.facebook.com/intern/diff/D20077321)

[ghstack-poisoned]

vishwakftw reviewed (approved):
Looks good to me, thanks!

Commit:
Update on "[JIT] add support for torch.lu to torchscript"


Fix for https://github.com/pytorch/pytorch/issues/33381, partial fix of https://github.com/pytorch/pytorch/issues/30786

Differential Revision: [D20077321](https://our.internmc.facebook.com/intern/diff/D20077321)

[ghstack-poisoned]

Commit:
Update on "[JIT] add support for torch.lu to torchscript"


Fix for https://github.com/pytorch/pytorch/issues/33381, partial fix of https://github.com/pytorch/pytorch/issues/30786

Differential Revision: [D20077321](https://our.internmc.facebook.com/intern/diff/D20077321)

[ghstack-poisoned]

Commit:
Update on "[JIT] add support for torch.lu to torchscript"


Fix for https://github.com/pytorch/pytorch/issues/33381, partial fix of https://github.com/pytorch/pytorch/issues/30786

Differential Revision: [D20077321](https://our.internmc.facebook.com/intern/diff/D20077321)

[ghstack-poisoned]

Commit:
Update on "[JIT] add support for torch.lu to torchscript"


Fix for https://github.com/pytorch/pytorch/issues/33381, partial fix of https://github.com/pytorch/pytorch/issues/30786

Differential Revision: [D20077321](https://our.internmc.facebook.com/intern/diff/D20077321)

[ghstack-poisoned]

Commit:
Update on "[JIT] add support for torch.lu to torchscript"


Fix for https://github.com/pytorch/pytorch/issues/33381, partial fix of https://github.com/pytorch/pytorch/issues/30786

Differential Revision: [D20077321](https://our.internmc.facebook.com/intern/diff/D20077321)

[ghstack-poisoned]
